cloud-controller-manager/do/resources_test.go

```diff
@@ -37,7 +37,207 @@ import (
+	}
+}
+
+var (
+	clusterID    = "0caf4c4e-e835-4a05-9ee8-5726bb66ab07"
+	clusterIDTag = buildK8sTag(clusterID)
+)
+
+func TestResourcesController_SyncResources(t *testing.T) {
+	tests := []struct {
+		name              string
+		dropletsSvc       godo.DropletsService
+		lbsSvc            godo.LoadBalancersService
+		err               error
+		expectedResources *resources
+	}{
+		{
+			name: "happy path",
+			dropletsSvc: &fakeDropletService{
+				listFunc: func(ctx context.Context, opt *godo.ListOptions) ([]godo.Droplet, *godo.Response, error) {
+					return []godo.Droplet{{ID: 2, Name: "two"}}, newFakeOKResponse(), nil
+				},
+			},
+			lbsSvc: &fakeLBService{
+				listFn: func(ctx context.Context, opt *godo.ListOptions) ([]godo.LoadBalancer, *godo.Response, error) {
+					return []godo.LoadBalancer{{ID: "2"}}, newFakeOKResponse(), nil
+				},
+			},
+			// both droplet and lb resources updated
+			expectedResources: &resources{
+				dropletIDMap:      map[int]*godo.Droplet{2: {ID: 2, Name: "two"}},
+				dropletNameMap:    map[string]*godo.Droplet{"two": {ID: 2, Name: "two"}},
+				loadBalancerIDMap: map[string]*godo.LoadBalancer{"two:": {ID: "two"}},
+			},
+		},
+		{
+			name: "droplets svc failure",
+			dropletsSvc: &fakeDropletService{
+				listFunc: func(ctx context.Context, opt *godo.ListOptions) ([]godo.Droplet, *godo.Response, error) {
+					return nil, newFakeNotOKResponse(), errors.New("droplets svc fail")
+				},
+			},
+			lbsSvc: &fakeLBService{
+				listFn: func(ctx context.Context, opt *godo.ListOptions) ([]godo.LoadBalancer, *godo.Response, error) {
+					return []godo.LoadBalancer{{ID: "2"}}, newFakeOKResponse(), nil
+				},
+			},
+			err: errors.New("droplets svc fail"),
+			// only lb resources updated
+			expectedResources: &resources{
+				dropletIDMap:      map[int]*godo.Droplet{1: {ID: 1, Name: "one"}},
+				dropletNameMap:    map[string]*godo.Droplet{"one": {ID: 1, Name: "one"}},
+				loadBalancerIDMap: map[string]*godo.LoadBalancer{"two:": {ID: "two"}},
+			},
+		},
+		{
+			name: "lbs svc failure",
+			dropletsSvc: &fakeDropletService{
+				listFunc: func(ctx context.Context, opt *godo.ListOptions) ([]godo.Droplet, *godo.Response, error) {
+					return []godo.Droplet{{ID: 2, Name: "two"}}, newFakeOKResponse(), nil
+				},
+			},
+			lbsSvc: &fakeLBService{
+				listFn: func(ctx context.Context, opt *godo.ListOptions) ([]godo.LoadBalancer, *godo.Response, error) {
+					return nil, newFakeNotOKResponse(), errors.New("lbs svc fail")
+				},
+			},
+			err: errors.New("lbs svc fail"),
+			// only droplet resources updated
+			expectedResources: &resources{
+				dropletIDMap:      map[int]*godo.Droplet{2: {ID: 2, Name: "two"}},
+				dropletNameMap:    map[string]*godo.Droplet{"two": {ID: 2, Name: "two"}},
+				loadBalancerIDMap: map[string]*godo.LoadBalancer{"one": {ID: "one"}},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
This is missing a `test := test` variable capture (like in line 354) before the call to `t.Run()` (which is important when doing `t.Parallel()`).